lib/tokenize.cpp

```diff
@@ -136,6 +136,14 @@ const Token * Tokenizer::isFunctionHead(const Token *tok, const std::string &end
             tok = tok->next();
         if (Token::Match(tok, "= 0|default|delete ;"))
             tok = tok->tokAt(2);
+        if (Token::simpleMatch(tok, "requires")) {
+            for (tok = tok->next(); tok && !Token::Match(tok, ";|{"); tok = tok->next()) {
+                if (tok->link() && Token::Match(tok, "<|[|("))
+                    tok = tok->link();
+                if (Token::simpleMatch(tok, "bool {"))
```

Comment:
Should this be `"%bool% {"`? But apparently it's not needed at all...

Reply:
No, this is for checking when doing explicit conversion to bool as the requires clause doesnt allow implicit conversion to bool so then the user will need to explicitly convert with either `static_cast<bool>(...)`, `(bool) ...`, `bool(...)` or `bool{...}`. However, the latter(ie `bool{}`) will exit the loop on `{` so this extra check is there to continue with the explicit conversion.

Reply:
If it is needed for something, we should have a test for that.

Reply:
Added test.